pennylane_lightning/lightning_qubit.py

```diff
@@ -38,8 +37,11 @@
 from pennylane.measurements import MeasurementProcess, Expectation, State
 from pennylane.wires import Wires
 
-# Remove after the next release of PL
-# Add from pennylane import matrix
+ABC_ARRAY = np.array(list(ABC))
+
+# tolerance for numerical errors
+tolerance = 1e-10
```

Comment:
I'm curious if there is any reason to set the tolerance as `1e-10`. The minimal positive normal value that fp32 supports if I remember correctly is around `1e-38` or so.

Reply:
At this first moment, we are following [PennyLane](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/devices/default_qubit.py#L37) lead here.
In the future, we can adjust this if needed.

If you are wondering why they choose this number, what I have to say is that this is a reasonably small number, that will be used to check the normalization of the statevector ([here](https://github.com/PennyLaneAI/pennylane-lightning/blob/96cff21e3df9a182eb1c283618683c3c26b3aa23/pennylane_lightning/lightning_qubit.py#L271)). 
In a case like that, we are summing, sometimes, millions of numbers and we can have a propagation of errors inconsistent with the fp32 precision.

Reply:
Thanks @AmintorDusko for the explanation and the nice work!